WordPress/gutenberg
# Global Styles: allow read access to users with `edit_posts` capabilities

Builds upon the base idea in:

- https://github.com/WordPress/gutenberg/pull/64797

With a view to:

- resolve https://github.com/WordPress/gutenberg/issues/64755
- make baby steps towards https://github.com/WordPress/gutenberg/issues/64670

## What

Allows any role that can edit a post, including custom post types, to read global styles from the API and in the editor.

## Why

Block style variations and other global styles are not available to non-admin editors. See https://github.com/WordPress/gutenberg/issues/64755

Furthermore, having global styles available in the post editor will, one day, allow block controls to reflect any values inherited from the theme/global styles. See: https://github.com/WordPress/gutenberg/issues/64670

### How

- Updating `read` capabilities to `edit_posts` in the global styles CPT (via filter)
- In the global styles API, updating permissions checks for GET requests to test for `current_user_can( 'edit_posts' )`
- In the global styles provider, remove any restrictions surrounding reading base config styles, and doing a "can update" check on the user global styles to either return the global styles editedEntity (for admins) or entity.


## Testing Instructions
1. As admin, make some changes to global styles in the site editor and save. This ensure there's a global styles CPT available.
2. Switch to an editor or author role.
3. Create a new post.
4. Run `await wp.data.resolveSelect( 'core/block-editor').getSettings()` in the browser console.
5. In the response, check the value of `Symbol(globalStylesDataKey)` - it should be a merged object of base theme and user global styles.
6. To check that you can't edit the global styles CPT, run `await wp.data.resolveSelect( 'core' ).getEntityRecord( 'root', 'globalStyles', await wp.data.resolveSelect( 'core' ).__experimentalGetCurrentGlobalStylesId(), { context: 'edit' } )` in the console. You should see a 403.
7. Now switch to your admin profile.
8. Edit global styles in the Site Editor. Now reset or roll back using global styles revisions.  Make sure it all works as expected. 


Run a test or two:

```
npm run test:unit:php:base -- --filter WP_REST_Global_Styles_Controller_Gutenberg_Test
```
